src/main/java/org/apache/aurora/scheduler/sla/SlaManager.java

```diff
@@ -445,8 +449,7 @@ private void incrementErrorCount(String prefix, String taskKey) {
   }
 
   private boolean skipSla(IScheduledTask task, long numActive) {
-    if (!tierManager.getTier(task.getAssignedTask().getTask()).isPreemptible()
-        && !tierManager.getTier(task.getAssignedTask().getTask()).isRevocable()) {
+    if (slaAwareKillNonProd || task.getAssignedTask().getTask().isProduction()) {
```

Comment:
The production flag is deprecated so we need to keep on using `isRevocable` and `isPreemptible` here.

Reply:
This is actually just a helper function introduced recently [here](https://github.com/apache/aurora/commit/c1791d6f94b7a98cdcecb3b70a8655f5320c3e58#diff-e6e423af634fb11298fbc8daa14eaa58R70) to help us cope with the fact that the migration to tiers was started but (in my opinion) never carried out in a straight forward manner.

So the code is doing exactly what you asked :slightly_smiling_face: 

I can change it back if you think it makes it easier to understand, however.

Reply:
That function is defined on `TierInfo` rather than `Task`. I think you are calling the legacy function here. If you build that back to the previous version that PR would be ready to merge from my perspective

Reply:
Woah! Fantastic catch 😄 . Fixing it right now and pushing the fix. Thanks